// react-paper/.module-cache/014dcfba549e373a198ab63ca7a39bf4af833057.js
/** @jsx React.DOM */

var React = require('react');

/*
 * Polymer paper-shadow can inject two elements from one components. We wrap em in one div.
 * TODO: Shadow z-levels
 */
var Shadow = React.createClass({displayName: 'Shadow',
  render: function() {
    return (
      React.DOM.div( {className:"shadow", style:this.shadowStyle}, 
        React.DOM.div( {className:"shadow-top", style:this.paperShadowTopStyle}),
        React.DOM.div( {className:"shadow-bottom", style:this.paperShadowBottomStyle})
      )
    );
  },

  shadowStyle: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    borderRadius: 'inherit',
    pointerEvents: 'none'
  },

  paperShadowTopStyle: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    borderRadius: 'inherit',
    pointerEvents: 'none',

    boxShadow: '0 2px 10px 0 rgba(0, 0, 0, 0.16)'
  },

  paperShadowBottomStyle: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    borderRadius: 'inherit',
    pointerEvents: 'none',

    boxShadow: '0 2px 5px 0 rgba(0, 0, 0, 0.26)'
  }
});

module.exports = Shadow;
