// react-paper/.module-cache/16312725babb2af4ec23553823f955ae6e37144a.js
/** @jsx React.DOM */
var React = require('react/addons');
var Progress = require('./Progress');

require('./css')('\
.slider-container {\
  position: relative;\
  height: 32px;\
  width: 100%;\
}\
\
.bar-container {\
  position: absolute;\
  top: 0;\
  left: 16px;\
  height: 100%;\
  width: 100%;\
  overflow: hidden;\
 }\
.bar-container > .progress {\
  position: absolute;\
  top: 15px;\
  left: 0;\
  height: 2px;\
  width: 100%;\
  padding: 8px 0;\
  margin: -8px 0;\
}\
.slider-knob {\
  display: -webkit-flex;\
  display: flex;\
  -webkit-align-items: center;\
  align-items: center;\
  -webkit-flex-direction: row;\
  flex-direction: row;\
  -webkit-justify-content: center;\
  justify-content: center;\
  -ms-touch-action: pan-y;\
  touch-action: pan-y;\
  position: absolute;\
  top: 0;\
  left: 0;\
  width: 32px;\
  height: 32px;\
}\
\
.transiting > .slider-knob {\
  transition: left 0.08s ease;\
}\
\
.slider-knob-inner {\
  moz-box-sizing: border-box;\
  border-radius: 50%;\
  background-color: #3f51b5;\
  box-sizing: border-box;\
  width: 12px;\
  height: 12px;\
  transition: height 0.18s ease, width 0.18s ease;\
}\
.slider-knob-inner:before {\
  content: "";\
  position: absolute;\
  top: 0;\
  left: 0;\
  width: 26px;\
  height: 26px;\
  margin-left: 3px;\
  border-radius: 50% 50% 50% 0;\
  -webkit-transform: rotate(-45deg) scale(0) translate(0);\
  transform: rotate(-45deg) scale(0) translate(0);\
  background-color: #3f51b5;\
}\
\
.slider-knob-inner:before, .slider-knob-inner:after {\
  transition: -webkit-transform .2s ease, background-color .18s ease;\
  transition: transform .2s ease, background-color .18s ease;\
}\
\
.slider-knob-inner:after {\
  content: attr(value);\
  position: absolute;\
  top: 0;\
  left: 0;\
  width: 32px;\
  height: 26px;\
  text-align: center;\
  color: #fff;\
  font-size: 10px;\
  -webkit-transform: scale(0) translate(0);\
  transform: scale(0) translate(0);\
}\
\
.slider-container.expand > .slider-knob > .slider-knob-inner:before {\
  -webkit-transform: rotate(-45deg) scale(1) translate(17px, -17px);\
  transform: rotate(-45deg) scale(1) translate(17px, -17px);\
}\
\
.slider-container.expand > .slider-knob > .slider-knob-inner:after {\
  -webkit-transform: scale(1) translate(0, -17px);\
  transform: scale(1) translate(0, -17px);\
}\
');

var Slider = React.createClass({displayName: 'Slider',
  render: function() {
    var min = this.props.min;
    var max = this.props.max;
    var value = this.props.value;

    var containerClass = React.addons.classSet({
      'slider-container': true,
      transiting: false,
      expand: false
    })

    return (
      React.DOM.div( {className:containerClass}, 
        React.DOM.div( {onClick:this.handleBarClick, className:"bar-container"}, 
          Progress( {max:max, value:value} )
        ),
        React.DOM.div( {onclick:this.handleKnob, style:{left: value + '%'}, className:"slider-knob"}, 
          React.DOM.div( {className:"slider-knob-inner"}

          )
        )
      )
    );
  },

  handleKnob: function() {
    console.log('knob', arguments)
  },

  handleBarClick: function() {
    console.log('bar', arguments)
  }
});

module.exports = Slider;
