// react-paper/.module-cache/314fb646d738c1396abc1ecdb8f3440c3d12aa7d.js
/** @jsx React.DOM */

var React = require('react/addons');

var Shadow = require('./Shadow');

var Dialog = React.createClass({displayName: 'Dialog',

  render: function() {
    var open = this.props.open || false;

    return (
      React.DOM.div( {style:open ? this.openDialogStyle : this.closedDialogStyle, className:"dialog"}, 
        Shadow(null ),
        React.DOM.div(null,  " I am a dialog! " )
      )
    );
  },

  openDialogStyle: {
    opacity: 1,
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    position: 'absolute',
    margin: '10px',
    padding: '10px',
    transform: 'none',
    WebkitTransform: 'none',
    width: '50%'
  },

  closedDialogStyle: {
    opacity: 0,
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    transform: 'scale(0.9) translateY(200%)',
    WebkitTransform: 'scale(0.9) translateY(200%)',
    position: 'absolute',
    margin: '10px',
    padding: '10px',
    width: '50%'
  }
});

module.exports = Dialog;
