// react-paper/.module-cache/851b8ba92013ee051e5b55d019088b353321736d.js
/** @jsx React.DOM */

var React = require('react/addons');
var Shadow = require('./Shadow');

require('./css')('\
.dialog {\
  z-index: 100;\
  background: white;\
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1);\
  transition: transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1);\
  position: absolute;\
  top: 0;\
  margin: 10px;\
  padding: 10px;\
  opacity: 0;\
  -webkit-transform: scale(0.9) translateY(200%);\
  transform: scale(0.9) translateY(200%);\
}\
\
.dialog.open {\
  opacity: 1;\
  -webkit-transform: none;\
  transform: none;\
}\
\
.dialog-content {\
  display: -webkit-flex;\
  display: flex;\
  -webkit-flex-direction: column;\
  flex-direction: column;\
  -webkit-align-items: center;\
  align-items: center;\
}\
');

var Dialog = React.createClass({displayName: 'Dialog',

  componentWillMount: function() {
    if (navigator.userAgent.indexOf("Safari") > -1)
      this.contentStyle.display = '-webkit-flex';
  },

  render: function() {
    var open = this.props.open || false;

    var dialogClass = React.addons.classSet({
      dialog: true,
      open: open
    });

    return (
      React.DOM.div( {className:dialogClass}, 
        Shadow(null ),
        React.DOM.div( {className:"dialog-content"}, 
          this.props.children
        )
      )
    );
  },

  contentStyle: {
    display: 'flex',
    flexDirection: 'column',
    WebkitFlexDirection: 'column',
    alignItems: 'center',
    WebkitAlignItems: 'center'
  },

  openDialogStyle: {
    zIndex: 100,
    background: 'white',
    opacity: 1,
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransform: 'none',
    transform: 'none',
    position: 'absolute',
    top: 0,
    margin: '10px',
    padding: '10px',
  },

  closedDialogStyle: {
    zIndex: 100,
    background: 'white',
    opacity: 0,
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransform: 'scale(0.9) translateY(200%)',
    transform: 'scale(0.9) translateY(200%)',
    position: 'absolute',
    top: 0,
    margin: '10px',
    padding: '10px',
  }
});

module.exports = Dialog;
