// react-paper/.module-cache/85c499bf567426e43661a122f214b6da464af1fe.js
/** @jsx React.DOM */

var React = require('react/addons');

var Shadow = require('./Shadow');

var Button = React.createClass({displayName: 'Button',
  render: function() {
    var raised = this.props.raised;
    if (this.props.background)
      this.buttonStyle.backgroundColor = this.props.background;
    if (this.props.color)
      this.buttonStyle.backgroundColor = this.props.color;

    return this.transferPropsTo(
      React.DOM.div( {style:this.buttonStyle, className:"button"}, 
        raised ? Shadow(null ) : null,
        this.props.children
      )
    );
  },

  buttonStyle: {
    margin: '0 0.29em',
    padding: '0.7em 0.57em',
    minWidth: '5.14em',
    boxSizing: 'border-box',
    lineHeight: 'normal',
    display: 'inline-block',
    position: 'relative',
    border: 0,
    background: 'transparent',
    textAlign: 'center',
    font: 'inherit',
    textTransform: 'uppercase',
    outline: 'none',
    borderRadius: '3px',
    webkitUserSelect: 'none',
    userSelect: 'none',
    cursor: 'pointer'
  }
});

module.exports = Button;
