// react-paper/.module-cache/a5b9104e06ab98eefa36926ea9790db786c6cf83.js
/** @jsx React.DOM */

var React = require('react/addons');

var HeaderPanel = React.createClass({displayName: 'HeaderPanel',

  componentWillMount: function() {
    if (navigator.userAgent.indexOf("Safari") > -1)
      this.headerStyle.display = '-webkit-flex';
  },

  render: function() {
    var header = this.props.children[0];
    var contents = this.props.children.slice(1);
    return (
      React.DOM.div( {style:this.panelStyle, className:"panel"}, 
        header,
        React.DOM.div( {style:this.contentPanelStyle, className:"content-panel"}, 
          React.DOM.div( {style:this.contentContainerStyle, className:"content-container"}, 
            contents
          ),
          React.DOM.div( {style:this.shadowStyle, className:"shadow"})
        )
      )
    );
  },

  panelStyle: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    overflowY: 'auto',
    overflowX: 'hidden',

    display: 'flex',
    flexDirection: 'column',
    WebkitFlexDirection: 'column'
  },

  contentPanelStyle: {
    position: 'relative',

    display: 'flex',
    flexDirection: 'column',
    WebkitFlexDirection: 'column',
    flex: '1 1 0',
    WebkitFlex: '1 1 0'
  },

  contentContainerStyle: {
    position: 'relative',

    overflowY: 'auto',
    overflowX: 'hidden'
  },

  shadowStyle: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: '6px',
    boxShadow: 'inset 0px 5px 6px -3px rgba(0, 0, 0, 0.4)'
  }
});

module.exports = HeaderPanel;
