// react-paper/.module-cache/fe0d9ec836b283bb0678918fa39ea78e5ffe04a9.js
/** @jsx React.DOM */

var React = require('react/addons');

var Shadow = require('./Shadow');

var Dialog = React.createClass({displayName: 'Dialog',

  componentWillMount: function() {
    if (navigator.userAgent.indexOf("Safari") > -1)
      this.contentStyle.display = '-webkit-flex';
  },

  render: function() {
    var open = this.props.open || false;

    return (
      React.DOM.div( {style:open ? this.openDialogStyle : this.closedDialogStyle, className:"dialog"}, 
        Shadow(null ),
        React.DOM.div( {className:"dialog-content", style:this.contentStyle}, 
          this.props.children
        )
      )
    );
  },

  contentStyle: {
    display: 'flex',
    flexDirection: 'column',
    WebkitFlexDirection: 'column',
    alignItems: 'center',
    WebkitAlignItems: 'center'
  },

  openDialogStyle: {
    zIndex: 100,
    background: 'white',
    opacity: 1,
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransform: 'none',
    transform: 'none',
    position: 'absolute',
    top: 0,
    margin: '10px',
    padding: '10px',
  },

  closedDialogStyle: {
    zIndex: 100,
    background: 'white',
    opacity: 0,
    WebkitTransition: '-webkit-transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    transition: 'transform 0.2s cubic-bezier(0.4, 0, 0.2, 1), opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
    WebkitTransform: 'scale(0.9) translateY(200%)',
    transform: 'scale(0.9) translateY(200%)',
    position: 'absolute',
    top: 0,
    margin: '10px',
    padding: '10px',
  }
});

module.exports = Dialog;
